controllers/endpoints.js

```diff
@@ -1,24 +1,39 @@
 const fleetService = require('../services/fleet');
+const formatDate = require('../public/js/endpoint.js'); 
 
 async function getEndpoints(req, res) {
     try {
-        endpoints = await fleetService.listEndpoints();
+        // Fetch endpoints
+        const endpointsData = await fleetService.listEndpoints();
 
+        // Fetch scripts for each endpoint
+        const scriptsData = await fleetService.getScriptByEndpoint(endpointsData.hosts);
+
+        // Map scripts to their respective endpoints
+        const endpointsWithScripts = fleetService.mergeEndpointAndScripts(endpointsData.hosts, scriptsData)
+
+        // Apply search filter if present
+        let filteredEndpoints = endpointsWithScripts;
         if (req.query.search) {
             const searchQuery = req.query.search.toLowerCase();
-            endpoints.hosts = endpoints.hosts.filter((host) => {
+            filteredEndpoints = endpointsWithScripts.filter(host => {
                 return host.hostname.toLowerCase().includes(searchQuery) ||
                     host.primary_ip.toLowerCase().includes(searchQuery) ||
                     host.primary_mac.toLowerCase().includes(searchQuery);
             });
         }
-
-        res.render("endpoints.ejs", { endpoints: endpoints.hosts });
+        filteredEndpoints.forEach(endpoint => {
+            endpoint.formatted_last_scan = formatDate.formatDate(endpoint.last_scan);
```

Comment:
You are using a function which is on the `public/js` path, which belongs to the frontend.
It is not the best practice, you should move it to be backend only/frontend only

Reply:
moved to the same endpoint.js (instead of public/js/enpoint.js)